packages/attune/tests/smoke/main.rs

```diff
@@ -0,0 +1,465 @@
+                            println!("✅ Extracted GPG key ID from revocation certificate: {key_id}");
+                            
+                            // Clean up config file
+                            if let Err(e) = fs::remove_file(config_path) {
+                                eprintln!("⚠️  Warning: Could not clean up GPG config file: {e}");
+                            }
+                            
+                            // Set the GPG_KEY_ID environment variable
+                            sh.set_var("GPG_KEY_ID", &key_id);
+                            println!("✅ GPG_KEY_ID environment variable set");
+                            
+                            return key_id;
+                        }
+                    }
+                }
+            }
+            eprintln!("❌ Could not find GPG key ID in revocation certificate output:\n{combined_output}");
+            panic!("GPG key ID not found in generation output");
+        }
+        Err(e) => {
+            eprintln!("❌ GPG key generation failed: {e}");
+            panic!("GPG key generation failed. Make sure GPG is installed and configured.");
+        }
+    };
+}
+
+/// Test that the CLI binary exists and is executable.
+fn test_cli_binary_exists(config: &TestConfig) {
+    let sh = Shell::new().unwrap();
+    set_up_shell_env(&sh, config);
+
+    let cli_path = &config.cli_path;
+    let version_result = cmd!(sh, "{cli_path} --help").run();
+
+    match version_result {
+        Ok(_) => {
+            println!("CLI binary is accessible and executable");
+            println!("✅ Test completed successfully!\n");
+        }
+        Err(e) => {
+            eprintln!("❌ CLI binary test failed: {e}");
+            eprintln!("Make sure the CLI is built and the path is correct.");
+            eprintln!("Current CLI path: {}", config.cli_path);
+            eprintln!(
+                "You can set ATTUNE_CLI_PATH environment variable to specify the correct path."
+            );
+            panic!("CLI binary not accessible");
+        }
+    }
+}
+
+/// Test creating multiple repositories.
+fn test_repo_create_multiple(config: &TestConfig) {
+    let sh = Shell::new().unwrap();
+    set_up_shell_env(&sh, config);
+
+    // Create two different Debian repos for testing.
+    let repo_names = ["debian-test-repo-1", "debian-test-repo-2"];
+    let cli_path = &config.cli_path;
+
+    for repo_name in &repo_names {
+        println!("Creating Debian repo: {repo_name}");
+        
+        let create_result = cmd!(sh, "{cli_path} apt repository create {repo_name}").run();
+
+        match create_result {
+            Ok(_) => println!("✅ Repo '{repo_name}' creation command executed successfully"),
+            Err(e) => {
+                eprintln!("❌ Repo creation failed for '{repo_name}': {e}");
+                panic!("Repo creation failed. Check your environment variables and CLI binary path.");
+            }
+        }
```

Comment:
[optional] we do this pattern a lot; this sort of thing would be ideal to wrap in a declarative macro: https://doc.rust-lang.org/reference/macros-by-example.html

this is a similar one where i don't really expect non-engineers to bother learning this so only if you want nerd points.

Also, the reason we'd want to use a macro for this instead of a function is:
- the macro more easily produces the panic in the actual location, instead of in the function itself
- macros can accept variadic input (like format arguments)